datafusion/functions/src/macros.rs

```diff
@@ -59,6 +59,30 @@ macro_rules! export_functions {
     };
 }
 
+macro_rules! make_function {
+    // single vector argument (a single argument followed by a comma)
+    ($FUNC:ident, $DOC:expr, $arg:ident,) => {
+        #[doc = $DOC]
+        pub fn $FUNC($arg: Vec<datafusion_expr::Expr>) -> datafusion_expr::Expr {
+            super::$FUNC().call($arg)
+        }
+    };
+    // variadic arguments (zero or more arguments, without commas)
+    ($FUNC:ident, $DOC:expr, $($arg:ident)*) => {
+        #[doc = $DOC]
+        pub fn $FUNC($($arg: datafusion_expr::Expr),*) -> datafusion_expr::Expr {
+            super::$FUNC().call(vec![$($arg),*])
+        }
+    };
+    // special to get_field function in datafusion/functions/src/core/mod.rs
+    ($FUNC:ident, $DOC:expr, $arg:ident, $arg2:ident) => {
```

Comment:
This works fine, but I hate having special conditions in shared functions/macros. Only alternative I can currently think of is rewriting `get_field` and make both args of type `Expr`

Reply:
I don't agree to rewrite the second argument to Expr.
I think we can remove it out of macro and define it specially given it's specialty

Reply:
Thought about that, the only problem is how can we split the definition of `pub fn functions()`  between two different macros, without pulling it into it's own macro?

Reply:
Maybe move `functions` out of macro?